smartmin/views.py

```diff
@@ -420,6 +420,8 @@ class SmartTemplateView(SmartView, TemplateView):
 
 
 class SmartReadView(SmartView, DetailView):
+    slug_field = None
+    slug_url_kwarg = None
```

Comment:
These default to 'slug' in `SingleObjectMixin` which tries to fetch objects by both slug and pk. Our view classes only have one URL and so can only work by slug *or* pk. If a subclass wants to use a slug field, they should explicitly set these attributes, which then tells `get_url_pattern` below to generate a URL for the slug field.